CanDIG/playwright-runbook
# Feature/search page

# Ticket
[Front-end runbook tests: Search page (DIG-1775)](https://candig.atlassian.net/browse/DIG-1775)

# Description
This PR adds comprehensive end-to-end tests for the CanDIGv2 search page, enhancing test coverage to ensure robust UI behavior during user interactions.

Key tests implemented:

- Sidebar Functionality: Ensures seamless interactions with the search sidebar.
- Patient Data Component: Validates data population and accuracy for patient results.
- Data Visualization: Confirms correctness and usability of visualized data.
- Clinical Table: Verifies clinical data table population and formatting.
- Genomic Table: Tests genomic data output and its expected alignment with input search parameters.

# Types of Change(s)

# How to Run Tests
Make sure to setup your .env file

```
npm install
npx playwright install
npx playwright test ui-tests/search-page.ui.ts --workers=1 --ui --headed
```
